osquery/logger/plugins/aws_kinesis.cpp

```diff
@@ -64,15 +67,43 @@ Status KinesisLoggerPlugin::logString(const std::string& s) {
   return forwarder_->logString(s);
 }
 
+Status KinesisLoggerPlugin::logStatus(const std::vector<StatusLogLine>& log) {
+  return forwarder_->logStatus(log);
+}
+
+void KinesisLoggerPlugin::init(const std::string& name,
+                               const std::vector<StatusLogLine>& log) {
+  google::ShutdownGoogleLogging();
+  google::InitGoogleLogging(name.c_str());
+  logStatus(log);
+}
+
 Status KinesisLogForwarder::send(std::vector<std::string>& log_data,
                                  const std::string& log_type) {
-  size_t retry_count = 100;
-  size_t retry_delay = 3000;
+  size_t retry_count = kKinesisMaxRetryCount;
+  size_t retry_delay = kKinesisInitialRetryDelay;
   size_t original_data_size = log_data.size();
+
   // exit if we sent all the data
   while (log_data.size() > 0) {
     std::vector<Aws::Kinesis::Model::PutRecordsRequestEntry> entries;
-    for (const std::string& log : log_data) {
+    std::vector<int> valid_log_data_indices;
+
+    int i = 0;
```

Comment:
Since this is used in several locations beyond an iteration counter, can we name it something more meaningful?